cairo/ethereum/cancun/transactions.cairo

```diff
@@ -332,3 +343,91 @@ func recover_sender{
+            tempvar res = Transaction(
+                new TransactionStruct(
+                    legacy_transaction=LegacyTransaction(cast(0, LegacyTransactionStruct*)),
+                    access_list_transaction=access_list_transaction,
+                    fee_market_transaction=FeeMarketTransaction(
+                        cast(0, FeeMarketTransactionStruct*)
+                    ),
+                    blob_transaction=BlobTransaction(cast(0, BlobTransactionStruct*)),
+                ),
+            );
+            return res;
+        }
+        if (transaction_type == TransactionType.FEE_MARKET) {
+            tempvar new_bytes = Bytes(new BytesStruct(data=bytes.data + 1, len=bytes_len - 1));
+            let fee_market_transaction = decode_to_fee_market_transaction(new_bytes);
+            tempvar res = Transaction(
+                new TransactionStruct(
+                    legacy_transaction=LegacyTransaction(cast(0, LegacyTransactionStruct*)),
+                    access_list_transaction=AccessListTransaction(
+                        cast(0, AccessListTransactionStruct*)
+                    ),
+                    fee_market_transaction=fee_market_transaction,
+                    blob_transaction=BlobTransaction(cast(0, BlobTransactionStruct*)),
+                ),
+            );
+            return res;
+        }
+        if (transaction_type == TransactionType.BLOB) {
+            tempvar new_bytes = Bytes(new BytesStruct(data=bytes.data + 1, len=bytes_len - 1));
+            let blob_transaction = decode_to_blob_transaction(new_bytes);
+            tempvar res = Transaction(
+                new TransactionStruct(
+                    legacy_transaction=LegacyTransaction(cast(0, LegacyTransactionStruct*)),
+                    access_list_transaction=AccessListTransaction(
+                        cast(0, AccessListTransactionStruct*)
+                    ),
+                    fee_market_transaction=FeeMarketTransaction(
+                        cast(0, FeeMarketTransactionStruct*)
+                    ),
+                    blob_transaction=blob_transaction,
+                ),
+            );
+            return res;
+        }
+    }
+    if (cast(tx.value.legacy_transaction.value, felt) != 0) {
+        tempvar res = Transaction(
+            new TransactionStruct(
+                legacy_transaction=tx.value.legacy_transaction,
+                access_list_transaction=AccessListTransaction(
+                    cast(0, AccessListTransactionStruct*)
+                ),
+                fee_market_transaction=FeeMarketTransaction(cast(0, FeeMarketTransactionStruct*)),
+                blob_transaction=BlobTransaction(cast(0, BlobTransactionStruct*)),
+            ),
+        );
+        return res;
+    }
+    with_attr error_message("TransactionTypeError") {
+        assert 0 = 1;
+        // unreachable
+        tempvar res = Transaction(
```

Comment:
not sure what's the cleanest way in cairo to return nothing here

Reply:
just `ret`